test/finally.cc

```diff
@@ -37,48 +43,56 @@ TEST(Finally, Fail) {
   auto e = []() {
     return Just(42)
         >> Raise("error")
-        >> Finally([](expected<int, std::exception_ptr>&& expected) {
+        >> Finally([](expected<
+                       int,
+                       std::variant<Stopped, std::runtime_error>>&& expected) {
              return Just(std::move(expected));
            });
   };
 
-  expected<int, std::exception_ptr> result = *e();
+  expected<
+      int,
+      std::variant<Stopped, std::runtime_error>>
+      result = *e();
 
   ASSERT_FALSE(result.has_value());
 
-  EXPECT_THAT(
-      [&]() { std::rethrow_exception(result.error()); },
-      ThrowsMessage<std::runtime_error>(StrEq("error")));
+  CHECK(std::holds_alternative<std::runtime_error>(result.error()));
+
+  EXPECT_STREQ(std::get<std::runtime_error>(result.error()).what(), "error");
 }
 
 TEST(Finally, Stop) {
   auto e = []() {
     return Eventual<std::string>([](auto& k) {
              k.Stop();
            })
-        >> Finally([](expected<std::string, std::exception_ptr>&& expected) {
+        >> Finally([](expected<std::string, std::variant<Stopped>>&& expected) {
              return Just(std::move(expected));
            });
   };
 
-  expected<std::string, std::exception_ptr> result = *e();
+  expected<
+      std::string,
+      std::variant<Stopped>>
+      result = *e();
 
   ASSERT_FALSE(result.has_value());
 
-  EXPECT_THROW(
-      std::rethrow_exception(result.error()),
-      eventuals::StoppedException);
+  EXPECT_STREQ(
+      std::get<Stopped>(result.error()).what(),
```

Comment:
Please don't forget to make sure that `Stopped` does not inherit from `std::exception`, thanks!